### ISSUE TITLE
`org.ballerinalang.debugger.test.adapter.variables.VariableVisibilityTest#localVariableVisibilityTest` fails intermittently

### ISSUE THREAD

**Issue by MaryamZi** [ballerina-platform/ballerina-lang#34009]
**Description:**
$title, noticed for PR builds.

```
debugger-intg-test-suite > debugger-intg-tests > org.ballerinalang.debugger.test.adapter.variables.VariableVisibilityTest > localVariableVisibilityTest FAILED
    org.ballerinalang.test.context.BallerinaTestException: Error occurred when fetching debug hit variables
        at org.ballerinalang.debugger.test.utils.DebugTestRunner.fetchVariables(DebugTestRunner.java:499)
        at org.ballerinalang.debugger.test.adapter.variables.VariableVisibilityTest.localVariableVisibilityTest(VariableVisibilityTest.java:217)

        Caused by:
        java.util.concurrent.TimeoutException
            at java.base/java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1886)
            at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2021)
            at org.ballerinalang.debugger.test.utils.client.DAPRequestManager.variables(DAPRequestManager.java:188)
            at org.ballerinalang.debugger.test.utils.client.DAPRequestManager.variables(DAPRequestManager.java:182)
            at org.ballerinalang.debugger.test.utils.DebugTestRunner.fetchVariables(DebugTestRunner.java:494)
            ... 1 more
```

**Comment by lochana-chathura** [comment]
Got a timeout for  `org.ballerinalang.debugger.test.adapter.variables.VariableVisibilityTest > controlFlowVariableVisibilityTest` as well. 
```
debugger-intg-test-suite > debugger-intg-tests > org.ballerinalang.debugger.test.adapter.variables.VariableVisibilityTest > controlFlowVariableVisibilityTest FAILED
    org.ballerinalang.test.context.BallerinaTestException: Error occurred when fetching debug hit variables
        at org.ballerinalang.debugger.test.utils.DebugTestRunner.fetchVariables(DebugTestRunner.java:499)
        at org.ballerinalang.debugger.test.adapter.variables.VariableVisibilityTest.controlFlowVariableVisibilityTest(VariableVisibilityTest.java:164)

        Caused by:
        java.util.concurrent.TimeoutException
            at java.base/java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1886)
            at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2021)
            at org.ballerinalang.debugger.test.utils.client.DAPRequestManager.variables(DAPRequestManager.java:188)
            at org.ballerinalang.debugger.test.utils.client.DAPRequestManager.variables(DAPRequestManager.java:182)
            at org.ballerinalang.debugger.test.utils.DebugTestRunner.fetchVariables(DebugTestRunner.java:494)
            ... 1 more
```

Please find the log here where both tests have failed: https://github.com/ballerina-platform/ballerina-lang/runs/4332363752?check_suite_focus=true

**Comment by MaryamZi** [comment]
`org.ballerinalang.debugger.test.remote.BallerinaRunRemoteDebugTest#testSuspendOnBallerinaFileRun` also failed for https://github.com/ballerina-platform/ballerina-lang/pull/34299.

```
debugger-intg-test-suite > debugger-intg-tests > org.ballerinalang.debugger.test.remote.BallerinaRunRemoteDebugTest > testSuspendOnBallerinaFileRun FAILED
    org.ballerinalang.test.context.BallerinaTestException: Timeout expired waiting for matching logs in debug mode.
        at org.ballerinalang.test.context.BMainInstance.debugMain(BMainInstance.java:361)
        at org.ballerinalang.test.context.BMainInstance.debugMain(BMainInstance.java:400)
        at org.ballerinalang.debugger.test.remote.BallerinaRunRemoteDebugTest.testSuspendOnBallerinaFileRun(BallerinaRunRemoteDebugTest.java:69)
```

Any possibility of prioritizing fixing these? Since the build takes quite some time already?

**Comment by praveennadarajah** [comment]
@MaryamZi `testSuspendOnBallerinaFileRun` failure will be mostly fixed with https://github.com/ballerina-platform/ballerina-lang/pull/34335

**Comment by NipunaRanasinghe** [comment]
@MaryamZi @lochana-chathura Intermittent timeout failures should be fixed with #34339 and we are also working on some debugger improvement to reduce the variable loading time. We are planning to restore the current timeouts values once the aforementioned improvements are merged. Please feel free to reopen the issue if you'll still be having any intermittent failures even after #34339.
